pkg/yang/types_builtin.go

```diff
@@ -944,10 +945,9 @@ func (r YangRange) Less(i, j int) bool {
 	}
 }
 
-// Validate sorts r and returns an error if r has either an invalid range or has
+// Validate returns an error if r has either an invalid range or has
 // overlapping ranges.
 func (r YangRange) Validate() error {
-	sort.Sort(r)
```

Comment:
Hmm, now I'm thinking we should error out if https://golang.org/pkg/sort/#IsSorted returns false.
Also, could you add a `Sort` method for `YangRange`, and call that at `parseRanges`? Could you also add a short test with a couple test cases for the method?

This way if the user somehow manually creates a YangRange, he/she can call `.Sort` to allow it to be validated.

Reply:
I see. Done